util/codec/decimal.go

```diff
@@ -30,6 +30,13 @@ func EncodeDecimal(b []byte, dec *types.MyDecimal, precision, frac int) ([]byte,
 	return b, errors.Trace(err)
 }
 
+func sizeDecimal(dec *types.MyDecimal, precision, frac int) int {
```

Comment:
Can we use `types.MyDecimalStructSize` as the estimated value?

Reply:
this method is port from MySQL and it seems can return an accurate value - -, but it's very complex..it's a good idea to use estimated value.